shell/platform/darwin/ios/framework/Source/SemanticsObject.h

```diff
@@ -19,6 +19,14 @@ constexpr float kScrollExtentMaxForInf = 1000;
 @class FlutterCustomAccessibilityAction;
 @class FlutterPlatformViewSemanticsContainer;
 
+@protocol FlutterSemanticsProtocol <NSObject>
+// Adding functionalities for any object to assign accessibilityContainer.
+// For example, this allows the PlatformView be able to link back to the accessibility tree.
```

Comment:
I'm confused by what this property is actually for. I see where it's set, but I don't see anything reading it except in tests. Am I missing some non-test behavior that's influenced by this?

Reply:
It is read in `FlutterTouchInterceptingView`: accessibilityContainer, it is a UIAccessibilty property that is read by the UIKit.

This protocol was created so that we can call this method to `FlutterTouchInterceptingView`. I initially added the method to the `FlutterTouchInterceptingView` so that `FlutterTouchInterceptingView` can access the container in its `accessibiltiyContainer` method.  

Then I discovered that SemanticsObjectTests is ARC, so we couldn't not include `FlutterTouchInterceptingView` (which uses MRC features) in the header of `SemanticsObjectTests`

This does make the code more confusing.